src/concrete_intersection.jl

```diff
@@ -536,8 +536,75 @@ function intersection(S::AbstractSingleton{N},
                      )::Union{Singleton{N}, EmptySet{N}} where {N<:Real}
     return invoke(intersection, Tuple{typeof(S), LazySet{N}}, S, P)
 end
+
 function intersection(P::AbstractPolytope{N},
                       S::AbstractSingleton{N}
                      )::Union{Singleton{N}, EmptySet{N}} where {N<:Real}
     return invoke(intersection, Tuple{typeof(S), LazySet{N}}, S, P)
 end
+
+"""
+    intersection(cup::UnionSet{N}, X::LazySet{N}) where {N<:Real}
+
+Return the intersection of a union of two convex sets and another convex set.
+
+### Input
+
+- `cup` -- union of two convex sets
+- `X`   -- convex set
+
+### Output
+
+The union of the pairwise intersections, expressed as a `UnionSet`.
+If one of those sets is empty, only the other set is returned.
+"""
+function intersection(cup::UnionSet{N}, X::LazySet{N}) where {N<:Real}
+    return intersection(cup.X, X) ∪ intersection(cup.Y, X)
```

Comment:
i would expect `intersection` to return a concrete set, not a `UnionSet`, but ok it can be used as a first step.

Reply:
In general we do not have a concrete set type for that.
The only other option I see is to remove this method again.

Reply:
true. unless we we make further assumptions in the sets (eg. polytopes). ok for keeping it.